aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogLock.java

```diff
@@ -0,0 +1,143 @@
+    lockTableName = getRandomName();
+    S3FileIO fileIO = new S3FileIO(clientFactory::s3);
+    glueCatalog = new GlueCatalog();
+    AwsProperties awsProperties = new AwsProperties();
+    dynamo = clientFactory.dynamo();
+    glueCatalog.initialize(catalogName, testBucketPath, awsProperties, glue,
+        new DynamoLockManager(dynamo, lockTableName), fileIO);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    GlueTestBase.afterClass();
+    dynamo.deleteTable(DeleteTableRequest.builder().tableName(lockTableName).build());
+  }
+
+  @Test
+  public void testParallelCommitMultiThreadSingleCommit() {
+    int nThreads = 20;
+    String namespace = createNamespace();
+    String tableName = getRandomName();
+    createTable(namespace, tableName);
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    DataFile dataFile = DataFiles.builder(partitionSpec)
+        .withPath("/path/to/data-a.parquet")
+        .withFileSizeInBytes(1)
+        .withRecordCount(1)
+        .build();
+
+    List<AppendFiles> pendingCommits = IntStream.range(0, nThreads)
+        .mapToObj(i -> table.newAppend().appendFile(dataFile))
+        .collect(Collectors.toList());
+
+    ExecutorService executorService = MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(nThreads));
+
+    Tasks.range(nThreads)
+        .retry(10000)
+        .throwFailureWhenFinished()
+        .executeWith(executorService)
+        .run(i -> pendingCommits.get(i).commit());
+
+    table.refresh();
+    Assert.assertEquals("Commits should all succeed sequentially", nThreads, table.history().size());
+    Assert.assertEquals("Should have all manifests", nThreads, table.currentSnapshot().allManifests().size());
+  }
+
+  @Test
+  public void testParallelCommitMultiThreadMultiCommit() {
+    String namespace = createNamespace();
+    String tableName = getRandomName();
+    createTable(namespace, tableName);
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    String fileName = UUID.randomUUID().toString();
+    DataFile file = DataFiles.builder(table.spec())
+        .withPath(FileFormat.PARQUET.addExtension(fileName))
+        .withRecordCount(2)
+        .withFileSizeInBytes(0)
+        .build();
+
+    ExecutorService executorService = MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(2));
+
+    AtomicInteger barrier = new AtomicInteger(0);
+    Tasks.range(2)
+        .stopOnFailure()
+        .throwFailureWhenFinished()
+        .executeWith(executorService)
+        .run(index -> {
+          for (int numCommittedFiles = 0; numCommittedFiles < 10; numCommittedFiles++) {
+            while (barrier.get() < numCommittedFiles * 2) {
+              try {
+                Thread.sleep(10);
+              } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)